Document/Page/ContentTypePage.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+/*
+ * This file is part of the Integrated package.
+ *
+ * (c) e-Active B.V. <integrated@e-active.nl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Integrated\Bundle\PageBundle\Document\Page;
+
+use Symfony\Component\Validator\Constraints as Assert;
+
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+use Integrated\Bundle\ContentBundle\Document\ContentType\ContentType;
+use Integrated\Bundle\ContentBundle\Document\Channel\Channel;
+
+/**
+ * @author Johan Liefers <johan@e-active.nl>
+ *
+ * @ODM\Document(collection="content_type_page")
+ */
+class ContentTypePage
+{
+    /**
+     * @var string
+     * @ODM\Id(strategy="UUID")
+     */
+    protected $id;
+
+    /**
+     * @var string
+     * @ODM\String
+     * @Assert\NotBlank
+     * @Assert\Regex(
+     *     pattern="/{slug}/",
+     *     message="Url must contain {slug}"
+     * )
+     */
+    protected $path;
+
+    /**
+     * @var string
+     * @ODM\String
+     * @Assert\NotBlank
+     */
+    protected $controllerService;
+
+    /**
+     * @var string
+     * @ODM\String
+     * @Assert\NotBlank
+     */
+    protected $controllerAction;
+
+    /**
+     * @var string
+     * @ODM\String
+     * @Assert\NotBlank
+     */
+    protected $layout;
+
+    /**
+     * @var \DateTime
+     * @ODM\Date
+     */
+    protected $createdAt;
+
+    /**
+     * @var \DateTime
+     * @ODM\Date
+     */
+    protected $updatedAt;
+
+    /**
+     * @var bool
+     * @ODM\Boolean
+     */
+    protected $disabled = false;
+
+    /**
+     * @var Channel
+     * @ODM\ReferenceOne(targetDocument="Integrated\Bundle\ContentBundle\Document\Channel\Channel")
+     */
+    protected $channel;
+
+    /**
+     * @var ContentType
+     * @ODM\ReferenceOne(targetDocument="Integrated\Bundle\ContentBundle\Document\ContentType\ContentType")
+     */
+    protected $contentType;
+
+    /**
+     * @param ContentType $contentType
+     * @param Channel $channel
+     * @param string $layout
+     * @param null $path
+     */
+    public function __construct(ContentType $contentType, Channel $channel, $layout = 'default.html.twig', $path = null)
+    {
+        $this->setContentType($contentType);
+        $this->setChannel($channel);
+        $this->setLayout($layout);
+        if ($path) {
+            $this->setPath($path);
+        } else {
+            $this->setPath(sprintf('/content/%s/{slug}', $contentType->getId()));
```

Comment:
No, it can't be a route, because it is added to the database as route and is dynamic